Marcio/stream live - pcx-2698 (#2415)
cloudflare/cloudflare-docs: products/stream/src/content/stream-live/start-stream-live.md

---
order: 6
pcx-content-type: tutorial
---

# Start a live stream

You can start a live stream using the Stream Dashboard or the API. After you subscribe to Stream, you can create Live Inputs and begin sending your live video to Cloudflare Stream using RTMPS.

> ## Using the Dashboard

Reviewer: Maybe "**Use** the Dashboard"?

1. To start a live stream using the Stream Dashboard, click the **Live Inputs** tab.

  ![Start a live stream by choosing the Live Inputs tab](./live-inputs-dash.png)

1. Next, click **Create Live Input**.

  ![Create a Live Input](./create-live-input.png)

1. Give your Live Input a name and click **Create Live Input**.

  ![Give your Live Input a name](./live-input-name.png)

1. Once you have created a Live Input, you can retrieve the RTMPS URL and Key.

  ![Your Live Input's RTMPS URL and key](./rtmps-key.png)

Within seconds of you pushing your live stream to Cloudflare Stream, you should see the live video stream:

![Your live video stream](./live-video-stream.png)

> ## Using the API

Reviewer: Consider "Use".

> To start a live stream programmatically, simply make a `POST` request to the `/live_inputs` endpoint:

Reviewer: ```suggestion
To start a live stream programmatically, make a `POST` request to the `/live_inputs` endpoint:
```

```bash

> curl -X POST \ -H "Authorization: Bearer $TOKEN" \https://api.cloudflare.com/client/v4/accounts/$ACCOUNT/stream/live_inputs \--data '{"meta": {"name":"test stream 1"},"recording": { "mode": "automatic", "timeoutSeconds": 10 }}'

Reviewer: You should split lines after each `\`.

```

* When the mode property is set to `automatic`, it means the live stream will be automatically available for viewing using HLS/DASH. In addition, the live stream will be automatically recorded for later replays.
* The `timeoutSeconds` property specifies how long a live feed can be disconnected before it results in a new video being created.

A successful response will return information about the live input:

```json
{
  "uid": "f256e6ea9341d51eea64c9454659e576",
  "rtmps": {
    "url": "rtmps://live.cloudflare.com:443/live/",
    "streamKey": "MTQ0MTcjM3MjI1NDE3ODIyNTI1MjYyMjE4NTI2ODI1NDcxMzUyMzcf256e6ea9351d51eea64c9454659e576"
  },
  "created": "2021-09-23T05:05:53.451415Z",
  "modified": "2021-09-23T05:05:53.451415Z",
  "meta": {
    "name": "My Live Stream"
  },
  "status": null,
  "live": {
    "mode": "automatic",
    "requireSignedURLs": false
  }
}
```

## Managing live inputs

You can update live inputs by making a `PUT` request:

```bash

> curl -X PUT \ -H "Authorization: Bearer $TOKEN" \https://api.cloudflare.com/client/v4/accounts/$ACCOUNT/stream/live_inputs/:input_id \--data '{"meta": {"name":"test stream 1"},"recording": { "mode": "automatic", "timeoutSeconds": 10 }}'

Reviewer: Split lines after each `\`.

```

> You can delete live inputs by making a `DELETE` request:

Reviewer: ```suggestion
Delete a live input by making a `DELETE` request:
```
(Avoiding "You can".)

```bash

> curl -X DELETE \ -H "Authorization: Bearer $TOKEN" \https://api.cloudflare.com/client/v4/accounts/$ACCOUNT/stream/live_inputs/:input_id

Reviewer: Split lines after each `\`.

```

## Requirements and known limitations 

### Requirements

> * Stream Live requires input GOP duration (keyframe interval) to be between four to 10 seconds.

Reviewer: ```suggestion
* Stream Live requires input GOP duration (keyframe interval) to be between 4 to 10 seconds.
```
Use numerals in both cases (makes it consistent and readers can compare the values).

* Closed GOPs required. This means that if there are any B frames in the video, they should always refer to frames within the same GOP. This setting is default in most encoder software such as OBS.

> * Stream Live only supports H264 video and AAC audio codecs as inputs. This requirement does not apply to inputs that are relayed to Stream Connect outputs.

Reviewer: ```suggestion
* Stream Live only supports H.264 video and AAC audio codecs as inputs. This requirement does not apply to inputs that are relayed to Stream Connect outputs.
```
H.264 appeared before with a dot.

> ### Known limitations (will be solved in coming weeks without any changes required from you): 

Reviewer: "(will be solved in coming weeks without any changes required from you):" should probably be on the next line, and not on the heading, which is too long.

* MP4 download from videos created from live streaming is not yet available.
* Watermarks cannot yet be used with live videos.

> * Live videos does not yet work on older iOS versions (iOS 10 launched 2016 and below).

Reviewer: ```suggestion
* The live videos feature does not yet work on older iOS versions: iOS 10, launched in 2016, and below.
```

> * GOP duration (keyframe interval) should to be constant during a broadcast.

Reviewer: ```suggestion
* GOP duration (keyframe interval) should be constant during a broadcast.
```

> * Hardware video encoding on Apple devices not yet supported. When using encoder software such as OBS, x264 software encoding is required.

Reviewer: ```suggestion
* Hardware video encoding on Apple devices is not yet supported. When using encoder software such as OBS, x264 software encoding is required.
```
